services/parser-database/parser.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import hashlib
+
+# import datetime
+# import requests
+import json
+
+import slate  # pylint: disable=import-error
+
+import retriever  # pylint: disable=import-error
+# import database  # pylint: disable=import-error
+import keywordmock  # pylint: disable=import-error
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Document:
+    """Class for Regulations Documents"""
+    def __init__(self, jurisdiction, shahash, url, last_updated):
+        self.id = None
+        self.jurisdiction = jurisdiction
+        self.shahash = shahash
+        self.url = url
+        self.last_updated = last_updated
+
+    @staticmethod
+    def from_dict():
+        pass
+
+    def to_dict(self):
+        pass
+
+    def __repr__(self):
+        return f"Document(\
+                id={self.id}, \
+                jurisdiction={self.jurisdiction}, \
+                last_updated={self.last_updated}, \
+                url={self.url}, \
+                shahash={self.shahash}\
+            )"
+
+
+class Article:
+    """Class for storing articles.
+    """
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+articles_in_memory = []
+keywords_in_memory = {}
+
+
+def get_article_by_number(art_num):
+    for item in articles_in_memory:
+        if art_num == str(item["articleNumber"]):
+            return item
+    return None
+
+
+def get_articles_that_match_keywords(keywords_list):
+    """Returns articles that match the the given keyword(s)
+
+    Args:
+        keywords_list (list): Keyword(s) to look for
+
+    Returns:
+        list: articles that match such keyword(s)
+    """
+    to_return = []
+    for keyword in keywords_list:
+        if keyword in keywords_in_memory:
+            print(keyword)
+            to_return.append(keywords_in_memory[keyword])
+    return to_return
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+    while i < len(pdf_text) - 1:
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
```

Comment:
So, this would be a really opportune time to use re.split (regular expression split), https://stackoverflow.com/questions/2136556/in-python-how-do-i-split-a-string-and-keep-the-separators for two reasons.
1. It can replace this entire function
2. You don't have to re-build the entire article, which allows you to store it in the database as close to it's original form as possible.

I am happy to help you figure out how to write a regexp equivalent of this. They're pretty confusing the first time you use them.

Plus if you have a test for this function first, then we can be extra sure the regex works :)

Reply:
If you want to, I had written this one the other day:
_ART[ÍI]CULO.?\d*\.?-?_
You should probably not take my word for it as I only tested it for a few minutes but I think it works. If you need an explanation on what it does ping me.

Reply:
Yes, I know about the regex. I was implementing it but then had to redesign other things and time got in the way.

I also discussed this with Spencer and `re.match()` is probably a better fit. I would definitely need your help on this @anniefu .